renderdoc/driver/gl/wrappers/gl_texture_funcs.cpp

```diff
@@ -2022,7 +2022,7 @@ bool WrappedOpenGL::Serialise_glTextureImage1DEXT(SerialiserType &ser, GLuint te
 
   size_t subimageSize = GetByteSize(width, 1, 1, format, type);
 
-  SERIALISE_ELEMENT_ARRAY(pixels, subimageSize);
+  SERIALISE_ELEMENT_ARRAY(pixels, FIXED_COUNT(pixels ? subimageSize : 0));
```

Comment:
What was the assert you were hitting here? The array/buffer serialisation should handle a `NULL` pointer silently, just clamping the size to 0. The `FIXED_COUNT` macro is only intended for the case where the array size is truly fixed but has been lost from the type information, like a function parameter that's `float ColorRGBA[4]`.

Reply:
For example, I get this error for a `glTexImage2D(..., NULL)` call:
```
serialiser.h( 260) - Error   - Assertion failed: 'byteSize' == 0
```
And then what should I use instead of FIXED_COUNT?

Reply:
Ah, the problem is that I'm using `byteSize` there which is still the original length, instead of `count` which has been clamped to 0 if the pointer is `NULL`.

If there's a case where the count should be modified, it'd be preferable to have the logic happen separately (if necessary on a `if(ser.IsWriting())` path) and then just serialise the count as-is. Like I say though, it should just handle `NULL` arrays without problem just by assuming the length is 0.

Reply:
I fixed it.

Reply:
I'd prefer to fix it properly by fixing the assert in `serialiser.h` like I mentioned. I've pushed a commit that uses the correct count which should get rid of the assert firing when you pass `NULL`. Can you revert this part of the change and check that you don't hit any asserts with the latest code?

Reply:
I reverted it. Your fix works, the assert disappeared. Thanks